### ISSUE TITLE
Using Locale-Emulator with zip extracting program like 7zip to extract filenames correctly

### ISSUE THREAD

**Issue by plintX** [xupefei/Locale-Emulator#378]
Hi,

I was wondering if it is possible to use locale emulator with a program like 7zip to extract filenames in a .zip archive properly?

The problem is that using 7zip without changing the system locale on a .zip file results in corrupt filenames because the unicode extraction is not done properly. This can be solved by switching the system locale to the filename language, but is inconvenient, and why we have programs like Locale-Emulator.

So does it work if we can run Locale-Emulator with 7zip to do correct extraction of filenames on .zip archive?

Thanks

**Comment by xupefei** [comment]
Unfortunately no. Please use Bandizip instead.

**Comment by plintX** [comment]
Cheers. In case somebody stumbles across this:

Turns out my issue wasn't exactly to do with unicode extraction, but because some filenames were encoded in Shift_JIS format instead of japanese UTF-8. More info here: https://marcosc.com/2008/12/zip-files-and-encoding-i-hate-you/

I ended up switching to Linux, and following the guide here: https://allencch.wordpress.com/2013/04/15/extracting-files-from-zip-which-contains-non-utf8-filename-in-linux/

Some issues that I had following that guide included:
- zip extraction didn't work on my NTFS mount. Not sure if this is because of ntfs, or just because it was a foreign mount. Moved the zip file to my local mount (ext4) and guide worked successfully.
- LANG environment variable didn't set the correct locale information for p7zip. Needed to set "LC_TYPE=C" to get the correct behaviour on my distro (Arch). Can determine the correct behaviour by running `locale`.

Closing.
